batcher/aligned-sdk/src/communication/messaging.rs

```diff
@@ -55,178 +55,198 @@ pub async fn send_messages(
+        }
+        Ok(ResponseMessage::EthRpcError) => {
+            error!("Batcher experienced Eth RPC connection error");
+            Err(SubmitError::EthereumProviderError(
+                "Batcher experienced Eth RPC connection error".to_string(),
+            ))
+        }
+        Ok(ResponseMessage::InvalidPaymentServiceAddress(received_addr, expected_addr)) => {
+            error!(
+                "Batcher responded with invalid payment service address: {:?}, expected: {:?}",
+                received_addr, expected_addr
             );
+            Err(SubmitError::InvalidPaymentServiceAddress(
+                received_addr,
+                expected_addr,
+            ))
+        }
+        Ok(ResponseMessage::InvalidProof(reason)) => {
+            error!("Batcher responded with invalid proof: {}", reason);
+            Err(SubmitError::InvalidProof(reason))
+        }
+        Ok(ResponseMessage::CreateNewTaskError(merkle_root, error)) => {
+            error!("Batcher responded with create new task error: {}", error);
+            Err(SubmitError::BatchSubmissionFailed(
+                "Could not create task with merkle root ".to_owned()
+                    + &merkle_root
+                    + ", failed with error: "
+                    + &error,
+            ))
         }
         Ok(ResponseMessage::ProtocolVersion(_)) => {
-            return Err(SubmitError::UnexpectedBatcherResponse(
+            error!("Batcher responded with protocol version instead of batch inclusion data");
+            Err(SubmitError::UnexpectedBatcherResponse(
                 "Batcher responded with protocol version instead of batch inclusion data"
                     .to_string(),
-            ));
+            ))
         }
         Ok(ResponseMessage::BatchReset) => {
-            return Err(SubmitError::ProofQueueFlushed);
+            error!("Batcher responded with batch reset");
+            Err(SubmitError::ProofQueueFlushed)
         }
         Ok(ResponseMessage::Error(e)) => {
             error!("Batcher responded with error: {}", e);
-        }
-        Ok(ResponseMessage::CreateNewTaskError(merkle_root)) => {
-            return Err(SubmitError::BatchSubmissionFailed(
-                "Could not create task with merkle root ".to_owned() + &merkle_root,
-            ));
-        }
-        Ok(ResponseMessage::InvalidProof(reason)) => {
-            return Err(SubmitError::InvalidProof(reason));
+            Err(SubmitError::GenericError(e))
         }
         Err(e) => {
-            return Err(SubmitError::SerializationError(e));
+            error!("Error while deserializing batch inclusion data: {}", e);
+            Err(SubmitError::SerializationError(e))
+        }
+    }
+}
+
+// Used to match the message received from the batcher,
+// with the NoncedVerificationData you sent
+// This is used to verify the proof you sent was indeed included in the batch
+fn match_batcher_response_with_stored_verification_data(
```

Comment:
This function has an issue with the non-paying wallet. As the bather will include the non-paying wallet nonce, and so the verification_data won't be found returning `Err(SubmitError::InvalidProofInclusionData)`

Reply:
handled in commit `e513997` , it worked for me, please make sure to test it again